src/Xamarin.Android.Build.Tasks/Tasks/Aapt2Link.cs

```diff
@@ -212,33 +302,10 @@ string ExpandString (string s)
 				return s;
 		}
 
-		bool ExecuteForAbi (string cmd, string currentResourceOutputFile)
+		bool ExecuteForAbi (string [] cmd, string currentResourceOutputFile)
 		{
-			var output = new List<OutputLine> ();
-			var aaptResult = RunAapt (cmd, output);
-			var success = !string.IsNullOrEmpty (currentResourceOutputFile)
-				? File.Exists (Path.Combine (currentResourceOutputFile + ".bk"))
-				: aaptResult;
-			foreach (var line in output) {
-				if (!LogAapt2EventsFromOutput (line.Line, MessageImportance.Normal, success))
-					break;
-			}
-			if (!string.IsNullOrEmpty (currentResourceOutputFile)) {
-				var tmpfile = currentResourceOutputFile + ".bk";
-				// aapt2 might not produce an archive and we must provide
-				// and -o foo even if we don't want one.
-				if (File.Exists (tmpfile)) {
-					if (aaptResult) {
-						MonoAndroidHelper.CopyIfZipChanged (tmpfile, currentResourceOutputFile);
-					}
-					File.Delete (tmpfile);
-				}
-				// Delete the archive on failure
-				if (!aaptResult && File.Exists (currentResourceOutputFile)) {
-					File.Delete (currentResourceOutputFile);
-				}
-			}
-			return aaptResult;
+			apks.Add (currentResourceOutputFile, RunAapt (cmd, currentResourceOutputFile));
```

Comment:
Just like `tempFiles` does this need a lock around it? If you used the apk-per-abi feature?